core/internal/filestream/filestreamrequest.go

```diff
@@ -134,17 +135,73 @@ type FileStreamRequestReader struct {
 // to the caller if the reader is discarded; otherwise, the caller
 // must call [Next] to get the updated request (minus the data that
 // was consumed).
-func NewRequestReader(request *FileStreamRequest) *FileStreamRequestReader {
+//
+// requestSizeLimitBytes is an approximate limit to the amount of
+// data to include in the JSON request.
+func NewRequestReader(
+	request *FileStreamRequest,
+	requestSizeLimitBytes int,
+) *FileStreamRequestReader {
+	requestSizeApprox := 0
+
+	// Increases requestSizeApprox and returns the number of strings to
+	// send from the list.
+	addStringsToRequest := func(data []string) int {
+		for i, line := range data {
+			nextSize := requestSizeApprox + len(line)
+
+			if nextSize > requestSizeLimitBytes {
+				// As a special case, if the first line is larger than the limit
+				// and we're not sending any other data, send the line and hope
+				// it goes through.
```

Comment:
what happens if it fails?

Reply:
do you mean what happens if it's still too big? there's nothing we can do at that point. filestream will likely enter a doomed retry loop.

Reply:
i wish we could break it into chunks that the backend could handle. 
do you think we should capture these events?

Reply:
unfortunately the line is the smallest unit of change in the filestream API, so we can't send a line in parts

we don't have a way of knowing if a filestream HTTP request failed due to a payload that's too large (I don't know if we send HTTP 413 codes)

Reply:
yeah...